_components/input-mask/guidance/implementation.md

```diff
@@ -1,24 +1,84 @@
+{:.usa-content-list }
+The examples in the Preview section above are common input masks, but you can adapt the code to meet the needs of your project. In order to extend the functionality of the input mask, you must include a `placeholder` attribute as the visible input mask, using the underscore `_` to represent a character the user should enter. You may use punctuation literals like `[space]`, `-`, `/`, or `(` to help format your mask, but these literals will not be included in the final submitted value. _See a complete list of format characters below_.
+
+{:.usa-content-list }
+- **Add component classes.** The structure should include a base element with the class `usa-input-mask`. Inside of that base element there should be an input element with the class `usa-input-mask__field` and a message element (span or div) with the class `usa-input-mask__message`.
+- **Add a `mask` attribute to the input element.** The mask attribute is used to define the format of your mask. Use `9`s for numbers and `A`s for letters. For example, a typical SSN number format in the mask attribute should look like `999-99-9999`.
+- **Add a `placeholder` attribute to the input element.** This indicates the mask format. Use underscores `_` for character input placeholders combined with any valid format characters (see below) to customize your input mask. e.g. A US telephone `placeholder` attribute may look like `(___) ___-____`
+- **Add a `maxlength` attribute to the input element.** This will be used as the limit referenced in the message and for validation.
+- **Set the attribute `type` to numeric if your mask is all numbers.** This will display the numeric keypad instead of the full keyboard.
+
+{:.usa-content-list }
+The following is an example of `input` element attributes for a Social Security Number mask:
 
 {:.usa-content-list }
 ```
-placeholder="___ ___ ___"
-pattern="\d\d\d \d\d\d \d\d\d"
+aria-describedby="ssn-hint ssn-hint-info"
+class="usa-input usa-input-mask__field"
+id="ssn"
 inputmode="numeric"
+mask="999-99-9999"
+maxlength="11"
+name="ssn"
+placeholder="___-__-____"
+type="text"
 ```
 
 {:.usa-content-list }
 <h3>The input mask and alphanumeric characters</h3>
```

Comment:
It should be safe to remove the `usa-content-list` class from the header elements. Additionally, can you replace the `<h3>` elements and replace them with `###`?
```suggestion
### The input mask and alphanumeric characters
```